src/services/timerFunction.ts

```diff
@@ -0,0 +1,214 @@
+import moment, { Moment } from "moment";
+import {
+  addFinishedExecution,
+  addOngoingExecution,
+  getOngoingExecutions,
+  setOngoingExecution,
+  removeOngoingExecution,
+} from "./localStorage";
+import { ExecutionStatus, OngoingExecution, FinishedExecution } from "./types";
+
+/**
+ * Coordena o a manipulação de execuções.
+ *
+ * O objetivo é centralizar as funções relacionadas à execuções, de forma a
+ * não poluir as Screens que em que serão utilizadas e que mudanças necessárias
+ * sejam automaticamente propagadas a todas as telas que necessitem trabalhar com
+ * execuções.
+ *
+ */
+
+/** Função responsável por criar uma nova execução */
+export const createExecution = async (cardInfo: any) => {
+  let startTime: Moment = moment();
+  let strStartTime: string = startTime.format("YYYY-MM-DDTHH:mm:ss[Z]ZZ");
+  console.log(
+    "timerFunction.createExecution: Criando execução - Horário atual: ",
+    strStartTime
+  );
+
+  let newExecution: OngoingExecution = {
+    startTime: strStartTime,
+    elapsedTime: 0,
+    latestStartTime: startTime,
+    idPatient: cardInfo.idPatient,
+    role: cardInfo.role,
+    activity: cardInfo.activity,
+    currentState: ExecutionStatus.Uninitialized,
+  };
+  await addOngoingExecution(newExecution);
+  // Chamar add ao storage
+};
+
+/** Função responsável por iniciar contagem de tempo de uma execução
+ * Retorna true se foi possível iniciar, podendo determinar a criação de setInterval()
+ * para fazer update de todas as execuções
+ */
+
+export const initializeExecution = async (id: number) => {
+  // Usar get do storage por indice
+
+  let isInitialized: boolean;
+  let execution = await getSingleOngoingExecution(id);
+  if (execution === null) {
+    return false;
+  }
+  if (
+    execution.currentState === ExecutionStatus.Paused ||
+    execution.currentState === ExecutionStatus.Uninitialized
+  ) {
+    let startTime: Moment = moment();
+    let strTime: string = startTime.format("YYYY-MM-DDTHH:mm:ss[Z]ZZ");
+    console.log(
+      "timerFunction.initializeExecution: Inicializando execução - Horário atual: ",
+      strTime
+    );
+    if (execution.currentState === ExecutionStatus.Uninitialized) {
+      execution.startTime = strTime;
+    }
+
+    execution.latestStartTime = startTime;
+    execution.currentState = ExecutionStatus.Initialized;
+
+    // Usar set de storage
+    await setOngoingExecution(execution, id);
+    isInitialized = true;
+  } else {
+    console.warn("Execução já iniciada");
+    isInitialized = false;
+  }
+  return isInitialized;
+};
+/** Função responsável por pausar uma execução */
```

Comment:
```suggestion
/** 
* Função responsável por pausar uma execução e no momento, o elapsedTime é calculado ao pausar
* @param index, índice da execução a ser pausada no array
 */
```

Considere trocar de `id` pra `index` onde é esperado o índice do array, acho que fica mais fácil de entender o que está sendo feito.